twiliotutorial/views.py

```diff
@@ -55,3 +67,74 @@ def handle_digits(self, digits):
             response.hangup()
 
         return response
+
+
+class BeerFact(View):
+
+    def post(self, request):
+        response = voice_response.VoiceResponse()
+
+
+        beer = Beer()
+        mug = beer.get_beer_fact()
+        name = mug['name']
+        logging.debug(f"Beer info: {name}")
+        response.say("Hello! I am going to drop a dank beer on you.", voice=settings.VOICE)
+        gather = voice_response.Gather(
+            action="/beertext?beerid=" + mug['id'],
+            action_on_empty_result=False,
+            timeout=200,
+            numDigits=1,
+            finishOnKey='')
+        gather.say(f"Our beer today is {name}", voice=settings.VOICE)
+
+        if 'abv' in mug.keys():
+            abv = mug['abv']
+            gather.say(f"Coming in at {abv} percent.", voice=settings.VOICE)
+
+        if 'style' in mug.keys():
+            if 'description' in mug['style'].keys():
+                description = mug['style']['description']
+                gather.say(f"{description}", voice=settings.VOICE)
+        response.append(gather)
+        response.say("Wow! You made it to the end. Be excellent to each other.", voice=settings.VOICE)
+        response.hangup()
+
+        response = HttpResponse(response.to_xml())
+        return response
+
+class BeerText(View):
+
+    def post(self, request):
+
+        logging.info("OOhhh...Engagement.")
+        client = Client(settings.TWILIO_ACCOUNT_SID, settings.TWILIO_ACCOUNT_TOKEN)
+
+        request_beerid = parse_qs(request.META['QUERY_STRING'])['beerid'][0]
+        request_from_number = request.POST.get("From", None)
+        request_to_number = request.POST.get("To", None)
+
+        if request_beerid is None:
```

Comment:
Same as above here, business logic out of the post method and just call helper functions.

Reply:
Seems like the response is html which should work for xml, Twilio doesnt mind.

Reply:
Moved to beer.py